soccer/gameplay/plays/offense/distraction.py

```diff
@@ -0,0 +1,178 @@
+                        distraction.State.cetner_pass, 
+                        lambda: self.subbehavior_with_name('move half').is_done_running() and self.subbehavior_with_name('make striker stay').is_done_running() , 
+                        'go from setting up a close pass to making the pass towards the center of the field')
+
+        self.add_transition(distraction.State.cetner_pass,
+                        distraction.State.passing, 
+                        lambda: self.subbehavior_with_name('center pass').is_done_running(), 
+                        'after getting the ball to the center of the field, pass to the distracting robot')
+
+        self.add_transition(distraction.State.passing, 
+                        distraction.State.cross, lambda: 
+                        (self.has_subbehavior_with_name('distract pass') and self.subbehavior_with_name('distract pass').is_done_running()) or 
+                        (self.has_subbehavior_with_name('get close ball') and self.subbehavior_with_name('get close ball').is_done_running()), 
+                        'have the distracrting robot capture the ball and go to passing to the striker')
+
+        self.add_transition(distraction.State.passing,
+                        distraction.State.shoot, lambda: 
+                        (self.has_subbehavior_with_name('striker pass') and self.subbehavior_with_name('striker pass').is_done_running() ) or 
+                        (self.has_subbehavior_with_name('striker get close ball') and self.subbehavior_with_name('striker get close ball').is_done_running()),
+                        'if the ball is already close to the ball, then go from the striker captureing the ball to shooting the ball')
+
+        self.add_transition(distraction.State.cross,
+                        distraction.State.shoot, lambda: 
+                        self.subbehavior_with_name('pass to striker').is_done_running() or (self.has_subbehavior_with_name('capture 3') and 
+                        self.subbehavior_with_name('capture 3').is_done_running()) , 
+                        'go from the striker receiving the pass to shooting the ball')
+
+        self.add_transition(distraction.State.shoot,
+                        distraction.State.setup, lambda: 
+                        self.has_subbehavior_with_name('shooting') and self.subbehavior_with_name('shooting').is_done_running(), 
+                        'repeat')
+
+    def on_enter_setup(self):
+        #capture ball and get striker and distractor in position
+        self.add_subbehavior(skills.capture.Capture(), 'capture1', required = True)
+        self.add_subbehavior(skills.move.Move(self.striker_point), 'striker moves', required = False, priority = 10)
+        self.add_subbehavior(skills.move.Move(self.distraction_recieve_pass_point), 'distract moves', required = True)
+        #if the ball is too far away to pass across the field
+        self.ball_is_far = main.ball().pos.y < (0.4*constants.Field.Length)
```

Comment:
The 0.4*constants.Field.Length should probably be a constant up above. You can either put it in the constructor or above the Enum (Use all caps snake case if you do that)